### PR TITLE
Wiki controller tests

### PR DESCRIPTION
**Pull request by lalithr95**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] tests pass -- `rake test`
* [x] code has been rebased on top of latest master (check if another pull request was added recently, and please rebase)
* [x] pull request is descriptively named and, if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer

Thanks!

### CODE REVIEW

**Comment by lalithr95** [comment]
This PR mainly adds all the tests for `WikiController` and further changes where code breaks. Improvements to some of the tags are also done which were identified when new tests break. (mostly tag closing). Few validations were also added which break for edge cases typically nil validation.

File: app/views/wiki/revisions.html.erb [hunk 1]
```diff
@@ -1,79 +1,81 @@
...
+      var newer = <%= @node.revisions.length-1 %>; older = <%= @node.revisions.length-2 %>;
       $('#diff').html(diffString($('#body_'+older).html(),$('#body_'+newer).html()))
-      $('tr').removeClass('warning')
-      $('#row'+older).addClass('warning')
-      $('#row'+newer).addClass('warning')
-    }
-  });
-
-  </script>
+      $(document).keydown(function(e){ if (e.keyCode == 39) { next() }})
```
Comment (jywarren) [inline app/views/wiki/revisions.html.erb]: All this should eventually be moved to a revisions.js file, not inline. But no need in this commit. Thanks!

File: app/views/wiki/index.html.erb [hunk 2]
```diff
@@ -13,12 +13,12 @@
   <% elsif params[:action] == "liked" %>
 
     <h2>Highly liked wiki pages <small>collaborative documentation by contributors like you</small></h2>
-    <p>&nbsp;<a style="margin-bottom:4px;" href="/wiki/new" class="btn btn-default"><i class="fa fa-plus-circle"></i> Create a new wiki page</a> <span style="margin-top:4px;" class="pull-right"><a href="/wiki/">by recency</a> | <a href="/wiki/popular">by popularity</a> | <span style="color:#aaa">by # of likes</span></p>
+    <p>&nbsp;<a style="margin-bottom:4px;" href="/wiki/new" class="btn btn-default"><i class="fa fa-plus-circle"></i> Create a new wiki page</a> <span style="margin-top:4px;" class="pull-right"><a href="/wiki/">by recency</a> | <a href="/wiki/popular">by popularity</a> | <span style="color:#aaa">by # of likes</span></p></span>
```
Comment (jywarren) [inline app/views/wiki/index.html.erb]: Here, the `span` tag has to go inside the `p` tag -- the tests complain :-)

Thanks, good catch on missing close span!

**Comment by jywarren** [comment]
Hi, Lalith - just get a few warnings for improperly nested tags. I know at least two were actually due to my own tags being wrong, but could you fix the issues (I noted one in an inline comment) and re-submit? Thanks!

````
  closed at byte 10490, line 254
  attributes at open: {"style"=>"margin-top:4px;", "class"=>"pull-right"}
  text around open: "a new wiki page</a> <span style=\"margin-"
  text around close: "by # of likes</span></p></span>\n\n\n  <tab"
ignoring attempt to close p with div
  opened at byte 10177, line 254
  closed at byte 10719, line 271
  attributes at open: {}
  text around open: "ou</small></h2>\n    <p>&nbsp;<a style=\"m"
  text around close: "ble>\n\n\n\n  <hr />\n  \n</div><!--/span-->\n\n"
ignoring attempt to close p with div
  opened at byte 10177, line 254
  closed at byte 10746, line 274
  attributes at open: {}
  text around open: "ou</small></h2>\n    <p>&nbsp;<a style=\"m"
  text around close: "!--/span-->\n\n\n      </div><!--/row-->\n  "
ignoring attempt to close p with div
  opened at byte 10177, line 254
  closed at byte 10768, line 275
  attributes at open: {}
  text around open: "ou</small></h2>\n    <p>&nbsp;<a style=\"m"
  text around close: "div><!--/row-->\n    </div><!--/.fluid-co"
ignoring attempt to close p with body
  opened at byte 10177, line 254
  closed at byte 12682, line 310
  attributes at open: {}
  text around open: "ou</small></h2>\n    <p>&nbsp;<a style=\"m"
  text around close: ";\" />\n</footer>\n\n\n  </body>\n</html>\n"
ignoring attempt to close p with html
  opened at byte 10177, line 254
  closed at byte 12690, line 311
  attributes at open: {}
  text around open: "ou</small></h2>\n    <p>&nbsp;<a style=\"m"
  text around close: "footer>\n\n\n  </body>\n</html>\n"
````

File: app/views/wiki/index.html.erb (hunk 2 above)
Comment (lalithr95) [inline app/views/wiki/index.html.erb]: Ah I missed that! Will update it

**Comment by lalithr95** [comment]
All these warnings were due to minor p and span tag changes. Updated tags tests should work fine now.

**Comment by jywarren** [comment]
Excellent. All tests pass, no warnings. Merging. Thanks a million!
